teemaderegister-react/src/components/SignUp.jsx

```diff
@@ -0,0 +1,129 @@
+import { setDocTitle } from '../utils/Helpers'
+import { Link } from 'react-router-dom'
+import { Row, Col, Form, Input, Button, Checkbox, message } from 'antd'
+import { UserOutlined, LockOutlined, MailOutlined } from '@ant-design/icons'
+
+const FormItem = Form.Item
+
+const { func, object, shape, bool, string } = PropTypes
+
+const propTypes = {
+  initSignUp: func.isRequired,
+  location: object.isRequired,
+  signup: shape({
+    hasError: bool.isRequired,
+    error: shape({
+      message: string
+    }).isRequired,
+    signUpSuccess: bool.isRequired
+  }).isRequired,
+  triggerSignUp: func.isRequired
+}
+
+class SignUp extends React.Component {
+  constructor (props) {
+    super(props)
+    this.submit = this.submit.bind(this)
+    this.formRef = React.createRef()
+  }
+
+  UNSAFE_componentWillReceiveProps (nextProps) {
+    // You don't have to do this check first, but it can help prevent an unneeded render
+    if (nextProps.signup.hasError) {
+      message.error(nextProps.signup.error.message)
+    }
+  }
+
+  componentDidMount () {
+    setDocTitle('SignUp')
+  }
+
+  componentWillUnmount () {
+    this.props.initSignUp()
+  }
+
+  submit (values) {
+    const roles = { roles: ['student'] }
+    const result = { ...values, ...roles }
+    window.setTimeout(() => {
+      this.props.triggerSignUp(result).then(() => {
+        if (this.props.signup.signUpSuccess) {
+          this.formRef.current.resetFields()
+          message.success('Account created, you can now log in')
+        }
+      })
+    }, 1500)
+  }
+
+  render () {
+    const crumbs = [{ url: this.props.location.pathname, name: 'Sign Up' }]
+    return (
+      <div className = 'signup width--public-page'>
+        <Breadcrumbs crumbs={crumbs} />
+        <Row gutter={8}>
+          <Col span={8} />
+          <Col xs={24} sm={8}>
+            <Form ref={this.formRef} onFinish={this.submit} className='form--narrow'>
+              <h2 className='text-align--center'>
+                Sign up to <span className='emphisize'>Te</span>
+              </h2>
+              <FormItem name='firstName' className='name' rules={[
+                { required: true, message: 'Please input your name!' }
+              ]}>
+                <Input
+                  prefix={<UserOutlined />}
+                  type='name'
+                  placeholder='First name'
+                />
+              </FormItem>
+              <FormItem name='lastName' className='name' rules={[
+                { required: true, message: 'Please input your name!' }
+              ]}>
+                <Input
+                  prefix={<UserOutlined />}
+                  type='name'
+                  placeholder='Last name'
+                />
+              </FormItem>
+              <FormItem name='email' rules={[
+                { required: true, message: 'Please input your email!' }
+              ]}>
+                <Input autoFocus prefix={<MailOutlined />} placeholder='Email' />
```

Comment:
would make sense to move autofocus to first field in the form, e.g. first name

Reply:
✓